### PR TITLE
Refactoring of vaultmgr

### PR DESCRIPTION
**Pull request by giggsoff**
We have overcomplicated logic in vaultmgr. Let's move it to vault and split handlers according to persist type.

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>

### CODE REVIEW

**Comment by zedi-pramodh** [comment]
Thanks for doing this @giggsoff also if you get chance later please look at volumemgr too and see if we can refactor that.

**Review by eriknordmark — COMMENTED** [review]
@giggsoff did you remove any legacy code as part of this, or is it just creating the Interface so that the zfs vs ext4 handling is better structured?

**Comment by cshari-zededa** [comment]
@giggsoff I will get to this review by end of this week.

**Comment by giggsoff** [comment]
> @giggsoff did you remove any legacy code as part of this, or is it just creating the Interface so that the zfs vs ext4 handling is better structured?

I tried to avoid changing the code, and leave the logic for ext4 with legacy mode support. So yes, it is mostly about splitting handlers by filesystem type.

File: pkg/pillar/vault/handler_zfs.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+		}
+	}
+	return &status
+}
+
+// checkOperationalStatus returns nil if for vaultPath properties
+// for in ZFS have the following values:
+// mounted:yes keystatus:available encryption:aes-256-gcm;
+// else return err
+func (h *ZFSHandler) checkOperationalStatus(vaultPath string) error {
```
Comment (vk-en) [inline pkg/pillar/vault/handler_zfs.go:210]: Not so long ago I already updated this function, but did not dare to move it from vaultzfs.go to the zfs package.  I think it makes sense now.

File: pkg/pillar/vault/key.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+// stageKey is responsible for talking to TPM and Controller
+// and preparing the key for accessing the vault
+// returns function to unstage the key
+func stageKey(log *base.LogObject, cloudKeyOnlyMode, useSealedKey, tpmKeyOnlyMode bool, keyDirName string, keyFileName string) (func(), error) {
+	// Create a tmpfs file to pass the secret to fscrypt
+	if err := os.MkdirAll(keyDirName, 755); err != nil {
+		return nil, fmt.Errorf("error creating keyDir %s %v", keyDirName, err)
+	}
+
+	if _, _, err := execCmd("mount", "-t", "tmpfs", "tmpfs", keyDirName); err != nil {
```
Comment (vk-en) [inline pkg/pillar/vault/key.go:72]: Can we use base package for Exec() calls in this package?

Comment (eriknordmark) [reply pkg/pillar/vault/key.go:72]: There might even be an os.Mount() call.
But I wouldn't change that as part of this PR; better to keep the structural changes separate from other desirable changes.

**Review by eriknordmark — APPROVED** [review]
Kick off tests.

**Pull request merged**